File: Arrays/Python/BestTimeToBuyAndSell.py
def maxProfitBruteForce(self, prices):
    keys = [(i,j) for i in range(len(prices)) for j in range(len(prices)) if i!=j and j>i]
    profit = 0 
    for key in keys:
        prof = (prices[key[1]]-prices[key[0]])
        if (prof>profit):
            profit =prof
    return profit

def maxProfit(prices):
        currProfit, maxProfit = 0,0        
        for i in range(1, len(prices)):
            currTick = prices[i] - prices[i-1]
            currProfit = max(currTick, currTick + currProfit)
            maxProfit = max(currProfit, maxProfit)
        return maxProfit

File: Arrays/Java/Easy/BestTimeToBuyAndSell.java
/*

Problem should be called, best time to have bought
and sold a stock in the past lmao.


https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
*/

public class BestTimeToBuyAndSell{

    public static void main(String[] Args) {

        System.out.println("Problem: Find Max Profit");
        int[] arrayA = {7,1,5,3,6,4};

        System.out.println(maxProfit(arrayA));

    }

    public static int maxProfit(int[] prices) {
        
        
        int maxProf = 0;
        
        if (prices.length == 0) return maxProf;
        
        int buyAtPrice = prices[0];
        int profit = 0;
        
        for (int i = 1; i < prices.length; i++){
            if(buyAtPrice < prices[i]){
                profit = prices[i] - buyAtPrice;
                maxProf = Math.max(profit,maxProf);       
            }else{
                buyAtPrice = prices[i];
            }
            
        }
        
        return maxProf;
        
        
        
    }
}